tests/index.test.ts

```diff
@@ -52,53 +56,59 @@ const ganacheProvider = ganache.provider({
   },
 });
 
-const provider = new Web3(ganacheProvider as any);
+const web3provider = new Web3(ganacheProvider as any);
 
 const ethersProvider = new ethers.providers.Web3Provider(
   ganacheProvider as any
 );
 
-const fetcher = constructFetchFetcher(fetch);
+const fetchFetcher = constructFetchFetcher(fetch);
+const axiosFetcher = constructAxiosFetcher(axios);
 
 const signer = wallet.connect(ethersProvider);
 const senderAddress = signer.address;
 
-const contractCaller = constructEthersContractCaller(
+const ethersContractCaller = constructEthersContractCaller(
   {
     providerOrSigner: signer,
     Contract: ethers.Contract,
   },
   senderAddress
 );
 
-describe('ParaSwap SDK', () => {
-  let paraSwap: AllSDKMethods;
+const web3ContractCaller = constructEthersContractCaller(
+  {
+    providerOrSigner: signer,
+    Contract: ethers.Contract,
+  },
+  senderAddress
+);
 
-  beforeAll(async () => {
-    paraSwap = constructSDK({
+describe.each([
+  ['fetchFetcher', fetchFetcher],
+  ['axiosFetcher', axiosFetcher],
+])('ParaSwap SDK: fetching methods: %s', (testName, fetcher) => {
+  test('getBalance', async () => {
+    const { getBalance } = constructGetBalances({
       network,
       fetcher,
-      contractCaller,
     });
```

Comment:
Good approach.
But I think it would be better to `paraSwap = constructPartialSDK({fetcher}, ...functions to test)` in `beforeAll` or just outside of `test()`
So that we are closer to the real-world usage. And closer to what we advertise in Readme